File: FleetTrack/src/FleetTrack.Domain/Entities/Mission.cs
using FleetTrack.Domain.Enums;

namespace FleetTrack.Domain.Entities;

public class Mission : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public MissionStatus Status { get; set; }
    public MissionPriority Priority { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime? ActualStartDate { get; set; }
    public DateTime? ActualEndDate { get; set; }
    public double EstimatedDistance { get; set; }
    public double? ActualDistance { get; set; }

    // Navigation properties
    public Guid VehicleId { get; set; }
    public Vehicle Vehicle { get; set; } = null!;
    public Guid DriverId { get; set; }
    public Driver Driver { get; set; } = null!;
    public ICollection<Waypoint> Waypoints { get; set; } = new List<Waypoint>();
}

File: fleettrack-frontend/src/types/mission.ts
export enum MissionStatus {
  Planned = 0,
  Assigned = 1,
  InProgress = 2,
  Completed = 3,
  Cancelled = 4
}

export enum MissionPriority {
  Low = 0,
  Medium = 1,
  High = 2,
  Urgent = 3
}

export interface MissionDto {
  id: string;
  name: string;
  description: string;
  status: MissionStatus;
  priority: MissionPriority;
  startDate: string;
  endDate?: string;
  estimatedDistance: number;
  actualDistance?: number;
  vehicleId: string;
  vehicleRegistration: string;
  driverId: string;
  driverName: string;
}

export interface CreateMissionRequest {
  name: string;
  description: string;
  priority: MissionPriority;
  startDate: string;
  endDate?: string;
  estimatedDistance: number;
  vehicleId: string;
  driverId: string;
}

export interface UpdateMissionRequest {
  name: string;
  description: string;
  priority: MissionPriority;
  status?: MissionStatus;
  vehicleId?: string;
  driverId?: string;
  startDate?: string;
  endDate?: string;
  estimatedDistance?: number;
  actualDistance?: number;
}

export interface UpdateMissionStatusRequest {
  status: MissionStatus;
}
